source/MaterialXView/Viewer.cpp

```diff
@@ -261,9 +261,6 @@ void Viewer::setupLights(mx::DocumentPtr doc, const std::string& envRadiancePath
     // Create a new light handler
     _lightHandler = mx::HwLightHandler::create();
 
-    // Clear context light user data.
-    _genContext.popUserData(mx::HW::USER_DATA_LIGHT_SHADERS);
```

Comment:
Client should not need to know about clearing specific user data created internally. So using a clearUserData() method below instead, so clear all data.

Reply:
This is implicitly created data. I assume you still want to expose being able to clear explicitly created data. Guess we don't have a use case yet so can add if needed.